webapp/controller/ProductsOverviewPage.controller.js

```diff
@@ -22,6 +33,12 @@ sap.ui.define(
         oView.setModel(supplierModel.getModel(), "suppliers");
         oView.setModel(oProductsOverviewViewModel, "productsOverviewView");
 
+        this.oView = oView;
+        this.oViewModel = oView.getModel("productsOverviewView");
+        this.productsTable = this.byId("idProductsTable");
```

Comment:
Try not to pollute 'this'. You can create a getter for the table like this.getProductTable